R/generatePartialDependence.R

```diff
@@ -246,78 +190,213 @@ generatePartialDependenceData = function(obj, input, features,
       target = td$class.levels
     else
       target = td$positive
-  }  else
+  } else
     target = "Risk"
 
-  args = list(obj = obj, data = data, fun = fun, td = td, individual = individual,
-    bounds = bounds, ...)
-
-  if (length(features) > 1L & !interaction) {
-    out = lapply(features, function(x) {
-      args$features = x
-      if (derivative) {
-        args$bounds = NULL
-        out = parallelMap(doPartialDerivativeIteration, x = rng[[x]], more.args = args)
-        rng = as.data.frame(rng[[x]])
-        colnames(rng) = x
-        centerpred = NULL
-      } else {
-        rng = as.data.frame(rng[[x]])
-        colnames(rng) = x
-        args$rng = rng
-        out = parallelMap(doPartialDependenceIteration, i = seq_len(nrow(rng)), more.args = args)
-        if (!is.null(center) & individual)
-          centerpred = as.data.frame(doPartialDependenceIteration(obj, data, center[, x, drop = FALSE],
-            x, fun, td, 1, bounds = bounds, ...))
-        else
-          centerpred = NULL
-      }
-      if (!individual)
-        doAggregatePartialDependence(out, td, target, x, rng)
-      else
-        doIndividualPartialDependence(out, td, nrow(data), rng, target, x, centerpred)
-    })
-    out = setDF(rbindlist(out, fill = TRUE))
+  # estimation using mmpf::marginalPrediction
+  # derivative estimation using numDeriv::grad or numDeriv::jacobian
+  if (derivative) {
+    ## args = list(model = obj, data = data, uniform = uniform, aggregate.fun = fun,
+    ##   predict.fun = getPrediction, n = n, ...)
+    points = sapply(features, function(x) uniformGrid(data[[x]], n[1]), simplify = FALSE)
```

Comment:
If `derivative = TRUE` and `uniform = FALSE` we compute the derivatives for the points on a uniform grid, right?

Reply:
ah yea. i guess that ought to change....